### ISSUE TITLE
Enhancement: When isEqualTo(Object) fails, use isEqualToComparingFieldByField to compute diff

### ISSUE THREAD

**Issue by zuckel** [assertj/assertj-core#1429]
#### Summary
When isEqualTo(Object) fails, we try to create a diff based on the objects' toString method. This works in many cases, but the diff can quickly become hard to parse if the compared objects are large or contain certain types of values that interfere with the toString/diff syntax.

I propose to enrich diff messages based on the output of isEqualToComparingFieldByField(Object), which tells exactly which fields differ and how.

This could happen additionally to the existing diff message.

This could only happen in certain cases:
- when no toString method is declared
- when the involved Strings reach a certain length, e.g. 100 characters

#### Example
Instead of
```java
Expected:<Fields[[bergo,domino,tipi] ]> but was:<Fields[[ergo,domino,toast] ]>
```

it could be

```java
Expected:<Fields[[bergo,domino,tipi] ]> but was:<Fields[[ergo,domino,toast] ]>
Expecting values:
  <["bergo", "tipi"]>
in fields:
  <["fieldA", "fieldC"]>
but were:
  <["ergo", "toast"]>
```

**Comment by saurabh-ku** [comment]
This looks like an interesting feature, but i think it could get a bit overwhelming in deeply nested objects. Have you decided on how to format error messages in such cases?

**Comment by zuckel** [comment]
I am only proposing to use what is already computed in isEqualToComparingFieldByField(Object). Whether that supports nesting or not is another story.

**Comment by joel-costigliola** [comment]
Showing the diff is not always relevant as we don't know if all the fields were used in `equals`.
If one wants a diff then the way to go is use the [recursive comparison](https://assertj.github.io/doc/#assertj-core-recursive-comparison).
